### DISCUSSION TITLE
Views as pages

### DISCUSSION THREAD

**Discussion by rpjvanaert** [AvaloniaUI/Avalonia#8613] [Q&A]
So I have been looking around for a solution to using views as tabs. The idea is that there is a menu on some side of the mainwindow and that the content of the rest of the mainwindow depending on what the user clicks on the menu screen. Main window containing and changing another view in itself. I think it violates some MVVM ideas, but hopefully this sketches my idea. Does anyone have an idea to guide me?

Closest I got was TabControl+TabItem, but I would like to use a viewmodel with each tab then and some other performance concerns with that then.

**Comment by timunie** [comment]
I'd use a ListBox on the side and bind to SelectedItem in the main ContentControl. Then each page is a UserControl which needs to be fetched via DataTemplates, if all pages are different ViewModels.

**Reply by rpjvanaert** [reply]
Thanks a lot I'll see what I can do with it
